File: src/line/P001.java
package line;

import java.util.Scanner;
import java.util.StringTokenizer;

public class P001 {

    public static void main(String[] args){
        String input = new Scanner(System.in).nextLine().trim();
        final StringTokenizer tokenizer = new StringTokenizer(input);
        int m = 20_000;
        boolean isFinish = false;
        while (tokenizer.hasMoreTokens()) {
            int distance = Integer.parseInt(tokenizer.nextToken());
            // @todo Write your code here.
            if(!isFinish){
                if(4 <= distance && distance <= 178){
                    int c = cost(distance);
                    if(m-c < 0){
                        isFinish = true;
                    }else {
                        m -= c;
                    }
                }else{
                    isFinish = true;
                }
            }
        }

        System.out.println(m);
        // @todo Write your code here.
    }

    public static int cost(int distance){
        int m = 720;
        if(distance <= 40){
            return m;
        }else{
            m += ((distance-40)/8 + (distance % 8 == 0 ? 0 : 1 )) * 80;
        }
        return m;
    }
}

File: AlgorithmTraining/AlgorithmTraining/line/P001.cpp
#include <iostream>
// #include <cstdio>  // uncomment to use scanf/printf instead of cin/cout
#include <sstream>
#include <string>

using namespace std;

int cost(int distance){
    int m = 720;
    if(distance <= 40){
        return m;
    }else{
        m += ((distance-40)/8 + (distance % 8 == 0 ? 0 : 1 )) * 80;
    }
    return m;
}

int main() {
    string line;
    getline(cin, line);
    stringstream ss(line);
    int m = 20000;
    bool isFinish = false;
    for (int distance; !(ss >> distance).fail(); ) {
        // @todo Write your code here.
        if(!isFinish){
            if(4 <= distance && distance <= 178){
                int c = cost(distance);
                if(m-c < 0){
                    isFinish = true;
                }else {
                    m -= c;
                }
            }else{
                isFinish = true;
            }
        }
    }
    // @todo Write your code here.
    cout << m << endl;
    return 0;
}
